app.rb

```diff
@@ -29,8 +29,28 @@ class App < Sinatra::Base
   end
 
   post '/user-signup/email-notification' do
-    logger.info(request)
-    WifiUser::UseCase::SnsNotificationHandler.new(logger).handle(request)
+    whitelist_checker = WifiUser::UseCases::CheckIfWhitelistedEmail.new(
+      gateway: Common::Gateway::S3ObjectFetcher.new(
+        bucket: ENV.fetch('S3_SIGNUP_WHITELIST_BUCKET'),
```

Comment:
Would avoid letting a controller use `ENV`, and would lean towards a controller using the sinatra's settings instead.

However with the above comment, I don't think this will apply anymore

Reply:
Sounds good but propose this be done independent of this refactoring step